src/sentry/static/sentry/app/views/organizationIntegrations/index.tsx

```diff
@@ -132,110 +151,123 @@ class OrganizationIntegrations extends AsyncComponent {
     window.open(url, '_blank');
   };
 
-  // Rendering
+  onRemoveInternalApp = (app: SentryApp): void => {
+    const apps = this.state.orgOwnedApps.filter(a => a.slug !== app.slug);
+    removeSentryApp(this.api, app).then(
+      () => {
+        this.setState({orgOwnedApps: apps});
+      },
+      () => {}
+    );
+  };
+
+  //Returns 0 if uninstalled, 1 if pending, and 2 if installed
+  getInstallValue(integration: AppOrProvider) {
+    const {integrations, appInstalls} = this.state;
+    if (isSentryApp(integration)) {
+      const install = appInstalls.find(i => i.app.slug === integration.slug);
+      if (install) {
+        return install.status === 'pending' ? 1 : 2;
+      }
+      return 0;
+    }
+    return integrations.find(i => i.provider.key === integration.key) ? 2 : 0;
+  }
 
-  renderProvider(provider) {
+  sortIntegrations(integrations: AppOrProvider[]) {
+    return integrations
+      .sort((a, b) => a.name.localeCompare(b.name))
+      .sort((a, b) => this.getInstallValue(b) - this.getInstallValue(a));
+  }
+
+  // Rendering
+  renderProvider = (provider: IntegrationProvider) => {
+    //find the integration installations for that provider
+    const integrations = this.state.integrations.filter(
+      i => i.provider.key === provider.key
+    );
     return (
       <ProviderRow
         key={`row-${provider.key}`}
         data-test-id="integration-row"
         provider={provider}
         orgId={this.props.params.orgId}
-        integrations={provider.integrations}
+        integrations={integrations}
         onInstall={this.onInstall}
         onRemove={this.onRemove}
         onDisable={this.onDisable}
         onReinstall={this.onInstall}
         enabledPlugins={this.enabledPlugins}
+        newlyInstalledIntegrationId={this.state.newlyInstalledIntegrationId}
       />
     );
-  }
+  };
 
-  renderSentryApps(apps, key) {
+  //render either an internal or non-internal app
+  renderSentryApp = (app: SentryApp) => {
     const {organization} = this.props;
     const {appInstalls} = this.state;
 
+    if (app.status === 'internal') {
+      return (
+        <SentryApplicationRow
+          key={`sentry-app-row-${app.slug}`}
+          data-test-id="internal-integration-row"
+          api={this.api}
+          onRemoveApp={() => this.onRemoveInternalApp(app)}
+          organization={organization}
+          app={app}
+        />
+      );
+    }
+
     return (
       <SentryAppInstallations
-        key={`sentry-app-row-${key}`}
+        key={`sentry-app-row-${app.slug}`}
         data-test-id="integration-row"
         api={this.api}
         organization={organization}
         installs={appInstalls}
-        applications={apps}
+        applications={[app]}
```

Comment:
Passing in just one app makes `SentryAppInstallations` confusing now. If we are only passing in one app then there should always only be one install (per that org) so it seems like a lot of the logic in `SentryAppInstallations` is unneeded now. I think this is the only component that uses it

Reply:
@MeredithAnya I am confused about what you are saying. In the existing code, we only ever pass one app into this function. Are you saying we should refactor `SentryAppInstallations` to only accept a single app?

Reply:
huh I guess we are. So maybe this is left over from changing this so many times but to me it seems weird (and don't know why i didn't notice before) that if we are always only passing in one app, why we make it an array of one instead of just passing in the object. Same with the installs.

Reply:
@MeredithAnya I don't know why it takes an array. I will change `SentryAppInstallations` to accept only a single app.

Reply:
@scefali to be fair this isn't necessarily something that needs to be changed in this PR, but I do think is worth doing at some point